src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -94,6 +94,16 @@ public class SecurityTokenDescriptor
         /// </summary>
         public IDictionary<string, object> AdditionalInnerHeaderClaims { get; set; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="HashSet{T}"/> which contains the default header claims that should be excluded from the JWT header.
+        /// </summary>
+        /// <remarks>
+        /// The default header claims are 'alg', 'kid', 'x5t', 'enc', and 'zip'.
+        /// </remarks>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public ISet<string> ExcludedDefaultHeaderClaims { get; set; }
```

Comment:
In general, Exclude APIs are not common in Identity Model. What are the alternatives that we considered?

Reply:
I don't think any others were considered. I proposed this since only excluding `kid` and `x5t` was very targeted and iffy, and the only other option that comes to mind is allowcating a set with all the default header claims in every `SecurityTokenDescriptor` which can be modified as desired, but then you end up with those allocations.

Reply:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2991